components/ws-manager-api/core.proto

```diff
@@ -95,6 +95,9 @@ message StartWorkspaceResponse {
 
     // OwnerToken is the token of the workspace owner used for authentication
     string owner_token = 2;
+
+    // Success tells if the workspace started successfully or not
+    bool success = 3;
```

Comment:
this does not make sense. If the workspace start (as seen from the `StartWorkspace` function) was not successful the call would return an error

Reply:
From what I could understand, Server does not receive the error, it only receives `StartWorkspaceResponse` from the API. And at this moment `StartWorkspaceResponse` does not have any information telling if ws-manager was able to start the workspace or not.

Reply:
Hm. This covers only a small subset of possible "workspace start failures". From the top of my head a workspace can fail during the following activities during startup (once it has been created in the DB):
 - image-builder (and interaction with it): captured in this PR in `workspace-starter.ts`
 - communication with `ws-manager`: captured by this PR by adding the `success`
 - everything else that can go wrong after the call to `ws-manager` has been issued (content initialization, image pull, etc.): not covered

IMO the 1st and 3rd part are the most interesting ones. An alternative approach would be to scan the DB regularly for workspaces that failed (`status.failure`) and use that as basis for a metric. That would not be a "workspace start request counter", but far more interesting I think.